### DISCUSSION TITLE
Katana VS BBOT Web Spidering

### DISCUSSION THREAD

**Discussion by Sh4d0wHunt3rX** [blacklanternsecurity/bbot#1266] [General]
I'm comparing these two tools for web spidering. 

I used these two commands:

`katana -u https://www.myaccounting.it/ -o katana.txt` and pass results from `httpx -l katana.txt -o httpx.txt -mc 200`

`bbot -t https://www.myaccounting.it/ -m httpx -om asset_inventory -c web_spider_distance=4 web_spider_depth=4 web_spider_links_per_page=1000 omit_event_types=["DNS_NAME_UNRESOLVED","URL_UNVERIFIED"] url_extension_httpx_only=[]`

I only considered links from https://www.myaccounting.it/ . There are many links exist in httpx result, but not in bbot, A lot from links cointain /wp-json/ and embed. I got diff and attached the links here. I also removed blacklist extensions such as .css .

Is this because of my settings for depth?

[httpx.txt](https://github.com/blacklanternsecurity/bbot/files/14982308/httpx.txt)
[bbot.txt](https://github.com/blacklanternsecurity/bbot/files/14982309/bbot.txt)
[diff.txt](https://github.com/blacklanternsecurity/bbot/files/14982339/diff.txt)

**Comment by TheTechromancer** [comment]
Yes, you will not see links like `https://www.myaccounting.it/wp-json/wp/v2/categories/291` because the depth of that URL is 5, but you have your max depth set to 4.

**Comment by TheTechromancer** [comment]
Thanks for testing this btw. Our goal is to be the best web spidering tool, but most likely there are still a few improvements we need to make before it's fully up to speed with katana. Most of these will involve better URL extraction logic.

Something to note is that right now BBOT strips off the query string (everything after the `?`). But @liquidsec is experimenting with making that option configurable.

**Reply by Sh4d0wHunt3rX** [reply]
Hey thanks a lot, actually I just understood what depth really means with your help 🙏 I tried with depth 10 and bbot had all links.

And you just read my mind, I was going to ask why there is no URL with query string and you explained it, thank you ❤️

Also, I was wondering, if it's possible to set a crawl duration option on bbot same as katana?

` -ct, -crawl-duration value    maximum duration to crawl the target for (s, m, h, d) (default s)`

I guess maybe as bbot is recursive it's not possible though.

**Reply by TheTechromancer** [reply]
You can put a time limit on a bbot scan with the `timeout` command:

```bash
# send ctrl+c to bbot after 10 seconds
timeout -s INT 10 bbot -t tesla.com
```

**Reply by Sh4d0wHunt3rX** [reply]
Wow, that's awesome, thanks for this ❤️

**Reply by Sh4d0wHunt3rX** [reply]
Hey @TheTechromancer , is there any possibility to set timeout for individual modules? For example same as what you gave me, "send kill nuclei after 10 seconds" ? Because I don't want to finish the whole scan in some cases, crawling and new hostnames is important for me, but for examplee vhost is slow and I don't want it to run more than 1 hour.  🙏

**Reply by TheTechromancer** [reply]
That's a good idea. We've actually already implemented that for gowitness. It wouldn't be hard to do for nuclei. You wanna make an issue?

**Reply by TheTechromancer** [reply]
Also if vhost is misbehaving, it might be a bug. Can you include any information about the vhost issue?

**Reply by Sh4d0wHunt3rX** [reply]
No, I just said as an example, sorry 😁
